lighthouse-core/test/lib/sentry-test.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+jest.mock('raven');
+
+const raven = require('raven');
+const Sentry = require('../../lib/sentry');
+
+/* eslint-env jest */
+
+describe('Sentry', () => {
+  let configPayload;
+  let originalSentry;
+  let originalMathRandom;
+  let ravenConfigFn;
+  let ravenMergeContextFn;
+  let ravenCaptureExceptionFn;
+  let mathRandomFn;
+
+  beforeEach(() => {
+    configPayload = {
+      url: 'http://example.com',
+      flags: {enableErrorReporting: true},
+      environmentData: {},
+    };
+
+    originalSentry = {...Sentry};
```

Comment:
seems like sentry isn't modified anywhere?

Reply:
although, overriding Math.random is also kind of horrifying :) Maybe we should extract a `shouldSample()` function on `Sentry` and replace that in this test instead

Reply:
the Sentry object is mutated everytime you call `init`, I'll add a comment about this.

`shouldSample` works for me